virttest/qemu_vm.py

```diff
@@ -3862,13 +3862,17 @@ def create(
                     monitor = qemu_monitor.wait_for_create_monitor(
                         self, m_name, m_params, timeout
                     )
+                    # Add this monitor to the list
+                    self.monitors.append(monitor)
                 except qemu_monitor.MonitorConnectError as detail:
-                    LOG.error(detail)
-                    self.destroy()
-                    raise
-
-                # Add this monitor to the list
-                self.monitors.append(monitor)
+                    if self.process.is_alive():
+                        LOG.error(detail)
+                        self.destroy()
+                        raise
+                    else:
+                        status = self.process.get_status()
+                        output = self.process.get_output().strip()
```

Comment:
```suggestion
                        output = self.process.get_output().strip()
                        self.destroy()
```
I think we also need to destroy the VM here.

Reply:
updated